### PR TITLE
create a docker-compose pro for licensed users

### PR DESCRIPTION
**Pull request by HarshCasper**
## Description

Based on a discussion, this PR proposes two separate files:

- `docker-compose.yml` for community users
- `docker-compose-pro.yml` for licensed users

This PR would also involve a documentation change (which I will take care off).

### CODE REVIEW

**Review by whummer — APPROVED** [review]
LGTM 👍 

It's a bit unfortunate that docker-compose does not support conditional statements, otherwise we could have parameterized the image name based on whether the API key is configured or not.

I'm still a bit on the fence whether having a single docker-compose.yml would make sense, and adding a comment to the file that mentions the image name to use for Community/Pro, in the spirit of keeping the number of top-level files limited. But I get the appeal of having it ready and available in the repo, such that users can simply reference `docker-compose -f docker-compose-pro.yml ...`. Not a strong preference, as both solutions seems to be a compromise 🤷

File: docker-compose.yml [hunk 1]
```diff
@@ -7,14 +7,11 @@ services:
     ports:
       - "127.0.0.1:4566:4566"            # LocalStack Gateway
       - "127.0.0.1:4510-4559:4510-4559"  # external services port range
-      - "127.0.0.1:53:53"                # DNS config (only required for Pro)
-      - "127.0.0.1:53:53/udp"            # DNS config (only required for Pro)
       - "127.0.0.1:443:443"              # LocalStack HTTPS Gateway (only required for Pro)
```
Comment (thrau) [inline docker-compose.yml:10]: we could also remove this?

Comment (thrau) [inline docker-compose.yml:10]: guess we could also remove this line

File: docker-compose.yml [hunk 2]
```diff
@@ -7,14 +7,11 @@ services:
...
     environment:
       - DEBUG=${DEBUG-}
       - PERSISTENCE=${PERSISTENCE-}
```
Comment (thrau) [inline docker-compose.yml:13]: persistence is also a pro feature

**Review by thrau — CHANGES_REQUESTED** [review]
adding to waldemar's comment, i agree that it makes sense to keep the number of top-level files low. but i think having separate files makes sense.

in a second iteration, we could maybe think about adding a separate folder demonstrating different ways of starting localstack, which would include the docker-compose files.

**Pull request merged**